packages/@react-aria/test-utils/src/checkboxgroup.ts

```diff
@@ -0,0 +1,162 @@
+ */
+
+import {act, within} from '@testing-library/react';
+import {CheckboxGroupTesterOpts, UserOpts} from './types';
+import {pressElement} from './events';
+
+interface TriggerCheckboxOptions {
+  /**
+   * What interaction type to use when triggering a checkbox. Defaults to the interaction type set on the tester.
+   */
+  interactionType?: UserOpts['interactionType'],
+  /**
+   * The index, text, or node of the checkbox to toggle selection for.
+   */
+  checkbox: number | string | HTMLElement
+}
+
+export class CheckboxGroupTester {
+  private user;
+  private _interactionType: UserOpts['interactionType'];
+  private _checkboxgroup: HTMLElement;
+
+
+  constructor(opts: CheckboxGroupTesterOpts) {
+    let {root, user, interactionType} = opts;
+    this.user = user;
+    this._interactionType = interactionType || 'mouse';
+
+    this._checkboxgroup = root;
+    let checkboxgroup = within(root).queryAllByRole('group');
+    if (checkboxgroup.length > 0) {
+      this._checkboxgroup = checkboxgroup[0];
+    }
+  }
+
+  /**
+   * Set the interaction type used by the checkbox group tester.
+   */
+  setInteractionType(type: UserOpts['interactionType']): void {
+    this._interactionType = type;
+  }
+
+  /**
+   * Returns a checkbox matching the specified index or text content.
+   */
+  findCheckbox(opts: {checkboxIndexOrText: number | string}): HTMLElement {
+    let {
+      checkboxIndexOrText
+    } = opts;
+
+    let checkbox;
+    if (typeof checkboxIndexOrText === 'number') {
+      checkbox = this.checkboxes[checkboxIndexOrText];
+    } else if (typeof checkboxIndexOrText === 'string') {
+      let label = within(this.checkboxgroup).getByText(checkboxIndexOrText);
+
+      // Label may wrap the checkbox, or the actual label may be a sibling span, or the checkbox div could have the label within it
+      if (label) {
+        checkbox = within(label).queryByRole('checkbox');
+        if (!checkbox) {
+          let labelWrapper = label.closest('label');
+          if (labelWrapper) {
+            checkbox = within(labelWrapper).queryByRole('checkbox');
+          } else {
+            checkbox = label.closest('[role=checkbox]');
+          }
+        }
+      }
+    }
+
+    return checkbox;
+  }
+
+  private async keyboardNavigateToCheckbox(opts: {checkbox: HTMLElement}) {
+    let {checkbox} = opts;
+    let checkboxes = this.checkboxes;
+    checkboxes = checkboxes.filter(checkbox => !(checkbox.hasAttribute('disabled') || checkbox.getAttribute('aria-disabled') === 'true'));
+    if (checkboxes.length === 0) {
+      throw new Error('Checkbox group doesnt have any non-disabled checkboxes. Please double check your checkbox group.');
+    }
+
+    let targetIndex = checkboxes.indexOf(checkbox);
+    if (targetIndex === -1) {
+      throw new Error('Checkbox provided is not in the checkbox group.');
+    }
+
+    if (!this.checkboxgroup.contains(document.activeElement)) {
+      act(() => checkboxes[0].focus());
```

Comment:
Should we do this by default? It can sometimes cause behaviour that doesn't match real life. Maybe there should be an option to force focus, but otherwise we throw if they aren't in the checkbox group already?

Reply:
From feedback for other users and my own experience from converting tests in Quarry to use the test utils, there is often a lot of friction when people are integrating the utils into the flow of their existing test and expect that calling these "toggle/trigger" functions should just work, even if active focus is in a separate place in their app and the real life use case would be to press Tab 4x to get to the checkboxgroup/etc. I think ideally everyones tests would simulate the exact user flows but realistically I'd like there to be as little friction as possible in adoption.

Reply:
makes sense, i was on the fence about it anyways. if they have a test not working at some point down the line, i think we can suggest they alter the flow to a real user one  and see if it still happens